Kitodo-Query-URL-Import/src/main/java/org/kitodo/queryurlimport/QueryURLImport.java

```diff
@@ -130,17 +133,24 @@ public DataRecord getFullRecordById(String catalogId, String identifier) throws
         if (searchFieldMapping.keySet().containsAll(searchFields.keySet())) {
             // Query parameters for HTTP request
             LinkedHashMap<String, String> queryParameters = new LinkedHashMap<>(parameters);
-
             // Search fields and terms of query
-            LinkedHashMap<String, String> searchFieldMap = new LinkedHashMap<>();
-            for (Map.Entry<String, String> entry : searchFields.entrySet()) {
-                searchFieldMap.put(searchFieldMapping.get(entry.getKey()), entry.getValue());
-            }
+            LinkedHashMap<String, String> searchFieldMap = getSearchFieldMap(searchFields);
+
             try {
                 URI queryURL = createQueryURI(queryParameters);
-                String queryString = queryURL.toString();
-                queryString = queryString + "&startRecord=0&maximumRecords=" + rows + "&query="
-                        + createSearchFieldString(searchFieldMap);
+                String queryString = queryURL.toString() + "&";
+                if (Objects.nonNull(interfaceType)) {
+                    if (Objects.nonNull(interfaceType.getStartRecordString())) {
+                        queryString = queryString + interfaceType.getStartRecordString() + equalsOperand + "0&";
+                    }
+                    if (Objects.nonNull(interfaceType.getMaxRecordsString())) {
+                        queryString = queryString + interfaceType.getMaxRecordsString() + equalsOperand + rows + "&";
+                    }
+                    if (Objects.nonNull(interfaceType.getQueryString())) {
+                        queryString = queryString + equalsOperand;
+                    }
```

Comment:
How does the condition effect the execution of the concatenation?

Reply:
Well spotted, this is one place where I forgot to add the return value of `interfaceType.getQueryString()` itself to complete query string. Fixed it now.